src/SFML/Graphics/Shape.cpp

```diff
@@ -267,14 +267,13 @@ void Shape::updateTexCoords()
 
     for (std::size_t i = 0; i < m_vertices.getVertexCount(); ++i)
     {
-        const float xratio      = m_insideBounds.width > 0
-                                      ? (m_vertices[i].position.x - m_insideBounds.left) / m_insideBounds.width
+        const float xratio      = m_insideBounds.size.x > 0
```

Comment:
Sucks that we cannot express this as:

```cpp
const auto ratio = (m_vertices[i].position - m_insideBounds.position) / m_insideBounds.size;
```

Maybe this suggests that `m_insideBounds.size` should not be zero here, or that we should check that before entering the loop? Haven't thought about this deeply.

Reply:
I just added a commit simplifying this function.